oauth2_provider/oauth2_validators.py

```diff
@@ -335,7 +335,7 @@ def _get_token_from_authentication_server(
             try:
                 access_token = AccessToken.objects.select_related("application", "user").get(token=token)
```

Comment:
This should probably be
```python
access_token = AccessToken.objects.select_related().get_or_create(token=token, user=user, application=None, scope=scope, expires=expires)
```

instead of trying to catch `DoesNotExist` and then calling get_or_create. This should block any other thread trying to access the same token, I believe.

Reply:
Thanks @jdelic. Will test and report back.